File: typescript/docs_code_snippets/functions-update.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

// setup
const fnName = `docs-snippet-fn-${Date.now()}`;
await opper.call(fnName, { input: "hello", output_schema: z.object({ reply: z.string() }) });
// /setup

// --- docs ---
const fn = await opper.functions.get(fnName);
console.log(`Name: ${fn.name}, Hit count: ${fn.hit_count}`);
// --- /docs ---

// cleanup
await opper.functions.delete(fnName);

File: python/docs_code_snippets/functions-update.py
import time
from opperai import Opper

opper = Opper()

# Setup: create function via call
fn_name = f"docs-snippet-{int(time.time())}"
opper.call(fn_name, input="hello", output_schema={"type": "object", "properties": {"reply": {"type": "string"}}})

# --- docs ---
fn = opper.functions.get(fn_name)
# Update is done by calling with new schemas - the function auto-updates
result = opper.call(
    fn_name,
    input={"text": "Updated input"},
    output_schema={"type": "object", "properties": {"result": {"type": "string"}}},
)
print(result.data)
# --- /docs ---

# Cleanup
opper.functions.delete(fn_name)
